torchbenchmark/util/extra_args.py

```diff
@@ -113,8 +116,16 @@ def apply_decoration_args(model: 'torchbenchmark.util.model.BenchmarkModel', dar
             import torch
             model.add_context(lambda: torch.cuda.amp.autocast(dtype=torch.float16), stage=TEST_STAGE.FORWARD)
     elif dargs.precision == "amp_bf16":
-        import torch
-        model.add_context(lambda: torch.cpu.amp.autocast(dtype=torch.bfloat16))
+        if model.test == "eval":
```

Comment:
Can we add an assert before line 119? 
```suggestion
        assert model.device == "cpu", "amp_bf16 is only supported on cpu device."
        if model.test == "eval":
```

Reply:
Done.